bin/si/src/secret/shared.ts

```diff
@@ -44,15 +45,16 @@ export async function getOrCreateChangeSet(
   ctx: Context,
   changeSetsApi: ChangeSetsApi,
   workspaceId: string,
-  changeSetId: string | undefined,
+  changeSetIdOrName: string | undefined,
   operationName: string,
 ): Promise<ChangeSetResult> {
-  // If change set specified, use it
-  if (changeSetId) {
-    // TODO: Validate that the change set exists
-    // For now, just return it
+  // If change set specified, resolve name to ID if needed
+  if (changeSetIdOrName) {
+    ctx.logger.debug(`Resolving change set: ${changeSetIdOrName}`);
+    const resolvedId = await resolveChangeSet(workspaceId, changeSetIdOrName);
+    ctx.logger.debug(`Resolved to change set ID: ${resolvedId}`);
     return {
-      changeSetId,
+      changeSetId: resolvedId,
```

Comment:
```suggestion
    const changeSetId = await resolveChangeSet(workspaceId, changeSetIdOrName);
    return {
      changeSetId,
```

Then we don't have to rename it either :)